s3/src/main/scala/quasar/destination/avalanche/s3/AvalancheS3Config.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 Precog Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.destination.avalanche.s3
+
+import argonaut._, Argonaut._
+import cats.implicits._
+import java.net.{URI, URISyntaxException}
+import quasar.blobstore.s3.{
+  AccessKey,
+  Bucket,
+  Region,
+  SecretKey
+}
+import quasar.destination.avalanche.WriteMode, WriteMode._
+import scala.{Either, StringContext}
+import scala.Predef.String
+
+final case class ClusterPassword(value: String)
+
+// sealed trait Authorization
+// object Authorization {
+//   final case class Keys(accessKey: AccessKey, secretKey: SecretKey, region: Region) extends Authorization
+// }
+
+final case class BucketConfig(
+  bucket: Bucket,
+  accessKey: AccessKey,
+  secretKey: SecretKey,
+  region: Region)
```

Comment:
FYI, we generally double-indent function- and class-definition continuations. So these parameters should all be four-space indented.